File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int ans = 0;
        int left = 0;
        int right = 0;
        while (right < nums.size()) {
            int zeros_found = 0;
            while (right < nums.size()) {
                if (nums[right] == 0) {
                    zeros_found += 1;
                }
                while (zeros_found > 1) {
                    if (nums[left] == 0) {
                        zeros_found -= 1;
                    }
                    left++;
                }
                ans = max(ans, right - left);
                right += 1;
            }
        }
        return ans;
    }
};

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        ans = 0
        left = 0
        right = 0
        while right < len(nums):
            zeros_found = 0
            while right < len(nums):
                if nums[right] == 0:
                    zeros_found += 1
                while zeros_found > 1:
                    if nums[left] == 0:
                        zeros_found -= 1
                    left += 1
                ans = max(ans, right - left)
                right += 1
        return ans
        